### DISCUSSION TITLE
Static Files on YARP Server

### DISCUSSION THREAD

**Discussion by mattsmithdev** [dotnet/yarp#1712] [Q&A] [answered]
Using YARP 1.10 on NET6 Minimal API. I'm trying to serve static files from `wwwroot` in the same solution that hosts YARP. I have the static files middleware being called prior to YARP, like...
```   
app.UseStaticFiles();
app.MapReverseProxy();
```
...Problem is that YARP is still intercepting the static files calls to the html files under wwwroot and trying to forward them to the default route. If I remove the `app.MapReverseProxy();` line then the static files are served properly. I'm experiencing the same issue with Swagger, so don't know if that's related. My endpoints that are defined in the solution are fine and get served prior to YARP intercepting them. It's just the static files and Swagger that get intercepted.

**Comment by MihaZupan** [comment, marked as answer]
`UseStaticFiles` will only short-circuit the pipeline if no other route was found.
If you define a route in YARP that can match the request, the static files middleware will skip the request and let YARP handle it.

If you want static files to attempt to serve the request before falling back to YARP, you can swap the order of `UseStaticFiles` and `UseRouting`.
```c#
app.UseStaticFiles(); // The middleware runs before routing happens => no route was found

app.UseRouting();

app.MapReverseProxy();
```

**Reply by mattsmithdev** [reply]
Thanks, so much. That solved it.

**Reply by samsp-msft** [reply]
I think the problem is that UseRouting is implicitly added to the beginning of the pipeline if it's not explicitly added in code.

**Reply by kaykaul** [reply]
@MihaZupan Thank you 😍
@mattsmithdev Thank you for the question. This bugged me for quite some time.
